### PR TITLE
cluster package

### PR DESCRIPTION
**Pull request by henderiw**
this is the package for a workload cluster

### CODE REVIEW

File: cluster/Kptfile [hunk 1]
```diff
@@ -0,0 +1,12 @@
+apiVersion: kpt.dev/v1
+kind: Kptfile
+metadata:
+  name: nephio-workload-cluster02
```
Comment (johnbelamaric) [inline cluster/Kptfile]: change to "cluster" to match the package directory name.

File: cluster/README.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
+# nephio-workload-cluster
```
Comment (johnbelamaric) [inline cluster/README.md]: Change to "cluster". Or I kind of like "nephio-workload-cluster", so I would also be open to changing the dir to that. The reason is "cluster" could be ONLY the cluster, whereas the "nephio-workload-cluster" is supposed to convey that it includes fully registering it and bootstrapping it.

File: cluster/README.md [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+## Description
+
+Deploying this package to the Nephio management cluster will result in the
+provisioning of a workload cluster, associated repository, tokens, secrets,
+and other resources needed to fully register the new cluster with Nephio.
+
+To accomplish this, the package will deploy two additional packages to the
```
Comment (johnbelamaric) [inline cluster/README.md]: This readme is out of date, it now results in 2 to the mgmt repo, 2 to the staging repo, correct? We should make sure it's not lying.

File: cluster/pv-cluster.yaml [hunk 4]
```diff
@@ -0,0 +1,20 @@
+apiVersion: config.porch.kpt.dev/v1alpha1
+kind: PackageVariant
+metadata:
+  name: example-cluster
+spec:
+  upstream:
+    package: nephio-cluster-capi-kind
+    repo: blueprint-manifests
```
Comment (johnbelamaric) [inline cluster/pv-cluster.yaml]: I think repo and revision have to be changed here

File: cluster/pv-configsync.yaml [hunk 5]
```diff
@@ -0,0 +1,20 @@
+apiVersion: config.porch.kpt.dev/v1alpha1
+kind: PackageVariant
+metadata:
+  name: example-configsync
+spec:
+  upstream:
+    package: nephio-configsync
+    repo: blueprint-manifests
```
Comment (johnbelamaric) [inline cluster/pv-configsync.yaml]: I think repo and revision have to be changed here

File: cluster/pv-configsync.yaml [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+    package: example-configsync
+    repo: mgmt-staging
+  packageContext:
+    data:
+      clusterName: example
+  pipeline:
+    mutators:
+    - image: gcr.io/kpt-fn/set-annotations:v0.1.4
+      configMap:
+        nephio.org/cluster-name: example
```
Comment (johnbelamaric) [inline cluster/pv-configsync.yaml]: missing newline

File: cluster/pv-repo.yaml [hunk 7]
```diff
@@ -0,0 +1,20 @@
+apiVersion: config.porch.kpt.dev/v1alpha1
+kind: PackageVariant
+metadata:
+  name: example-repo
+spec:
+  upstream:
+    package: nephio-repo
+    repo: blueprint-manifests
```
Comment (johnbelamaric) [inline cluster/pv-repo.yaml]: I think repo and revision have to be changed here

File: cluster/pv-rootsync.yaml [hunk 8]
```diff
@@ -0,0 +1,20 @@
+apiVersion: config.porch.kpt.dev/v1alpha1
+kind: PackageVariant
+metadata:
+  name: example-rootsync
+spec:
+  upstream:
+    package: nephio-rootsync
+    repo: blueprint-manifests
```
Comment (johnbelamaric) [inline cluster/pv-rootsync.yaml]: I think repo and revision have to be changed here

File: cluster/pv-rootsync.yaml [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+    package: example-rootsync
+    repo: mgmt-staging
+  packageContext:
+    data:
+      clusterName: example
+  pipeline:
+    mutators:
+    - image: gcr.io/kpt-fn/set-annotations:v0.1.4
+      configMap:
+        nephio.org/cluster-name: example
```
Comment (johnbelamaric) [inline cluster/pv-rootsync.yaml]: missing newline

**Review by johnbelamaric — CHANGES_REQUESTED** [review]

File: cluster/Kptfile (hunk 1 above)
Comment (henderiw) [reply cluster/Kptfile]: fixed

File: cluster/README.md (hunk 2 above)
Comment (henderiw) [reply cluster/README.md]: ok changed the dir to nephio-workload-cluster

File: cluster/README.md (hunk 3 above)
Comment (henderiw) [reply cluster/README.md]: ok I updated the readme

File: cluster/pv-cluster.yaml (hunk 4 above)
Comment (henderiw) [reply cluster/pv-cluster.yaml]: changed to:
repo: nephio-example-packages
    revision: v1

File: cluster/pv-repo.yaml (hunk 7 above)
Comment (henderiw) [reply cluster/pv-repo.yaml]: repo: nephio-example-packages
    revision: v1

File: cluster/pv-configsync.yaml (hunk 6 above)
Comment (henderiw) [reply cluster/pv-configsync.yaml]: fixed

File: cluster/pv-rootsync.yaml (hunk 9 above)
Comment (henderiw) [reply cluster/pv-rootsync.yaml]: fixed

File: nephio-workload-cluster/README.md [hunk 10]
```diff
@@ -0,0 +1,15 @@
...
+Deploying this package to the Nephio management cluster will result in the
+provisioning of a workload cluster, associated repository, tokens, secrets,
+and other resources needed to fully register the new cluster with Nephio.
+
+To accomplish this, the package will deploy two additional packages to the management
+cluster repo (one for the cluster and one for the repository) and two additional packages
+to the management staging repo (one for configsync and one for rootsync).
+The names of those packages will be based on the name of the cluster-name. So, if this package
+is cloned to the name "cluster-01", it will deploy two additional packages,
+"cluster-01-nephio-workload-cluster", "cluster-01-repo", "cluster-01-rootsync" and "cluster-01-configsync"
```
Comment (johnbelamaric) [inline nephio-workload-cluster/README.md]: I think this first one would just be "cluster-01-cluster", right?

**Comment by vjayaramrh** [comment]
@henderiw are there any instructions on how to test this package? Thanks

File: nephio-workload-cluster/README.md (hunk 10 above)
Comment (henderiw) [reply nephio-workload-cluster/README.md]: yes sorry. Fixed

**Comment by henderiw** [comment]
@vjayaramrh I pasted the details in slack yesterday.

sig-auto-pkg-mgmt channel 6:03pm PST yesterday

**Comment by johnbelamaric** [comment]
https://nephio.slack.com/archives/C04RY9FSB9V/p1684857835165019

is the link...some of that will move to standard setup (Metal LB, for example). Metal LB is something we put in our "sandbox" but wouldn't be something we deploy when using "bring-your-own-cluster". Similarly for CAPI, in fact. So, we'll have to be sure we separate those things.

**Pull request merged**